pkg/scd/operational_intents_handler.go

```diff
@@ -396,11 +400,42 @@ func validateAndReturnUpsertParams(
 		return nil, stacktrace.NewError("Missing required UssBaseUrl")
 	}
 
+	valid.ussBaseURL = string(params.UssBaseUrl)
+
+	if params.SubscriptionId != nil {
+		valid.subscriptionID, err = dssmodels.IDFromOptionalString(string(*params.SubscriptionId))
+		if err != nil {
+			return nil, stacktrace.NewError("Invalid ID format for Subscription ID: `%s`", *params.SubscriptionId)
+		}
+	}
+
+	if params.NewSubscription != nil {
+		// The spec states that NewSubscription.UssBaseUrl is required and an empty value
+		// makes no sense, so we will fail if an implicit subscription is requested but the base url is empty
+		if params.NewSubscription.UssBaseUrl == "" {
+			return nil, stacktrace.NewError("Missing required USS base url for new subscription (in parameters for implicit subscription)")
+		}
+		if params.SubscriptionId != nil {
+			return nil, stacktrace.NewError("Cannot provide both a Subscription ID and request an implicit subscription")
+		}
+		valid.implicitSubscriptionRequested = true
+		valid.implicitSubscriptionBaseURL = string(params.NewSubscription.UssBaseUrl)
+		valid.implicitSubscriptionForConstraints = *params.NewSubscription.NotifyForConstraints
```

Comment:
`NotifyForConstraints` may be nil: that was checked before but not anymore: it should be added back.

Reply:
Good catch!